api/types.proto

```diff
@@ -884,3 +884,46 @@ message MaybeEncryptedRecord {
 	bytes data = 2;
 	bytes nonce = 3;
 }
+
+// PrivilegeProfiles specifies per-architecture security configuration/permissions
+message PrivilegeProfiles {
+	// Privileges required to run in Windows
+	oneof windows {
+		WindowsRawPrivilegeProfile windows_raw = 1;
+	}
+	// Privileges required to run in Linux
+	oneof linux {
+		LinuxRawPrivilegeProfile linux_raw = 2;
+	}
+}
+
+// WindowsRawPrivilegeProfile provides the low level privileges a windows container needs
+message WindowsRawPrivilegeProfile {
+	repeated Mount mounts = 1 [(gogoproto.nullable) = false];
+
+	// JSON string containing credential specs created with gMSA to use with managed service accounts in Windows
+	// (see https://blogs.msdn.microsoft.com/containerstuff/2017/01/30/create-a-container-with-active-directory-support/)
+	string credential_spec = 2;
+}
+
+
+// LinuxRawPrivilegeProfile provides the low level privileges a linux container needs
+message LinuxRawPrivilegeProfile {
+	message Device {
+		// The path to the device
+		string path = 1;
+		// The Read/Write/Mknod access for the device
+		string rwm = 2;
+	}
+
+	repeated Mount mounts = 1 [(gogoproto.nullable) = false];
+
+	// Linux capabilities required by the container
+	repeated string capabilities = 2;
```

Comment:
Are you proposing it actually being an enum? What would be the path for dealing with changes in capabilities?

Slightly concerned that inspecting profiles directly becomes a bit harder, but it will definitely be a lot more compressed on the wire this way.